### ISSUE TITLE
Amplify push fails on 11.0.4

### ISSUE THREAD

**Issue by dwil618** [aws-amplify/amplify-category-api#1400]
### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

16.x

### Amplify CLI Version

11.0.4

### What operating system are you using?

Tried onWindows/Linux

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No changes, just updated CLI

### Describe the bug

Build fails during `amplify push`. After the GraphQL operations are generated, the build fails, with error:

```

- Generating GraphQL operations
--
1135 | ✔ Generated GraphQL operations successfully and saved at src/graphql
1136 | - Generating
1137 | ✔ Code generated successfully and saved in file src/API.ts
1138 |  
1139 | ·[1A
1140 | ·[1A
1141 | ·[1A
1142 | ·[1A
1143 | ·[1A
1144 | ·[1A
1145 | ·[1A
1146 | ·[1A
1147 | ·[1A
1148 | - Uploading files.
1149 | ·[1A
1150 | ·[1A
1151 | ·[1A
1152 | ·[1A
1153 | ·[1A
1154 | ·[1A
1155 | ·[1A
1156 | ·[1A
1157 | ·[1A
1158 | ·[1A
1159 | - Uploading files.
1160 | - Uploading files.
1161 | 🛑 Cannot read property 'primaryKeyInfo' of undefined
1162 |  
1163 | Learn more at: https://docs.amplify.aws/cli/project/troubleshooting/
```

### Expected behavior

Build should work without error, as it does when running on 11.0.3 or lower

### Reproduction steps

Not sure which part of our build this error relates to. There are no errors for specific resources in the logs or CloudFormation stack events. We have the following backend resources:

 - GraphQL API
 - Cognito Auth (User Pool + Groups)
 - Amplify Hosting
 - Custom CDK stack (with just SSM parameters)

None of these have failed previously, and no changes have been made to the source code prior to the fails (still works on 11.03)

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by josefaidt** [comment]
Marking as a duplicate of https://github.com/aws-amplify/amplify-category-api/issues/1399

**Comment by dwil618** [comment]
Hi @AaronZyLee 

 - Yes we only saw this in the push step - init worked fine
 - Yes amplify hosting is in use

Looks like 11.0.4 has been taken down, so I can't run any tests with it to validate whether the codegen throws any errors.

**Comment by dwil618** [comment]
So `amplify codegen models` runs with no errors 

Got a slightly different output from the local run of `amplify push`:
```
✔ Generated GraphQL operations successfully and saved at src/graphql
✔ Code generated successfully and saved in file src/API.ts
Deployment state saved successfully.
🛑 Cannot read property 'primaryKeyInfo' of undefined
```


The error looks separate from the codegen process to me, as it happens after the deployment state save process which takes several seconds. Also the stack seems to have fully deployed, showing success in CloudFormation, but the command still fails.

**Comment by AaronZyLee** [comment]
@dwil618 Thanks for the response. While we are still digging into whether the version 11.0.3 is affected, it would be nice if you could provide more context.

- Do the command `amplify codegen model-introspection --output-dir .` fail for you? If it fails, does it fail on version 11.0.3?
- What is the value of the feature flag `respectPrimaryKeyAttributesOnConnectionField`? This can be found at the `amplify/cli.json`
- If possible, could you send a report by using `amplify diagnose` and the generated project identifier?

**Comment by phani-srikar** [comment]
Hi @dwil618, please upgrade your CLI to `11.0.5` or higher to not run into this codegen issue during `amplify push`.

**Comment by dwil618** [comment]
@phani-srikar This issue is resolved in `11.0.5` - thanks!
